### DISCUSSION TITLE
Ideas from a veteran CMMS user

### DISCUSSION THREAD

**Discussion by carlosvirreira** [Shelf-nu/shelf.nu#377] [Ideas]
@jurrejansen

Had a fantastic talk with one of our community members, Gustavo. 

He has worked with a handful of CMMS'es in the past, found shelf, and upon exploring it. Found refreshing concepts that he is excited to implement for his maintenance firm. 

Inventory report generation would be a good way to provide documents to customers when requested.

Annual reminders for asset upkeep like annual preventative maintenance schedules would be great.

Also mass asset addition or asset edit is a huge time saver when building databases or migrating from older systems.

Some way to link work orders/ support tickets and generate field service reports for said assets would be a huge plus for me.

**Comment by carlosvirreira** [comment]
My thoughts:

> Inventory report generation would be a good way to provide documents to customers when requested.
I would like to think of the most primitive report we can generate. With some possibilities to expand the reporting detailing further. 
What would be the most basic report? Go to Reporting > Select assets (or categories)  > Toggle what the report should cover (basically thee available fields on the UI > Export to PDF and send it per mail?)

> Annual reminders for asset upkeep like annual preventative maintenance schedules would be great.
This could happen on thee asset page itself or a separate area. < a bit scared of over cluttering the UI but, this feels like a similar problem to tackle with your 'batches' idea Jurre. Just like when we were thinking 

> Also mass asset addition or asset edit is a huge time saver when building databases or migrating from older systems.
This one I am not sure about the interaction. I would say that they can use a CSV import for this to populate their db. Ideally in a simple way so without much extra attributes just the minimum as highlighted on our CSV template.

> Some way to link work orders/ support tickets and generate field service reports for said assets would be a huge plus for me.
Not sure - we might need to interview Gustavo for this.


----

Should we tackle the first two points and then schedule a call to get more insight on the rest?

**Comment by jurrejansen** [comment]
**Inventory report generation would be a good way to provide documents to customers when requested.**

What does Gustav wants to see in a report? I think it would good to have some preconfigured reports based on user needs to start off with. Some examples down below.

- Assets per category
- Assets per status
- Assets in custody
- Assets per location
- Asset check-out/in over time

**Annual reminders for asset upkeep like annual preventative maintenance schedules would be great.**

As for Reminders / Alerts, could be an index where a user can create an entry and add assets that need to be related to whatever the Reminder / Alert is for. In the Reminder / Alert settings certain actions can be configured and of course the date/time the alert should trigger, if it should be repeated (monthly/annually), which users should be notified, etc.

**Also mass asset addition or asset edit is a huge time saver when building databases or migrating from older systems.**

For me this sounds more like multi-select + bulk action functionality. 

**Some way to link work orders / support tickets and generate field service reports for said assets would be a huge plus for me**

Indeed, need more information

**Comment by carlosvirreira** [comment]
October 5 at 17:00 hours we can talk with Gustavo. Please add to your agenda @jurrejansen
